superset/cli/main.py

```diff
@@ -50,9 +50,12 @@ def make_shell_context() -> Dict[str, Any]:
 ):
     module = importlib.import_module(module_name)
     for attribute in module.__dict__.values():
-        if isinstance(attribute, click.core.Command):
+        if isinstance(attribute, (click.core.Command, click.core.Group)):
             superset.add_command(attribute)
 
+            if isinstance(attribute, click.core.Group):
```

Comment:
We need to ensure that group commands aren't added as a top-level command. Regrettably there's no way of inspecting a (sub-)command to see if it's part of a group, but given how groups are defined, i.e., prior to the sub-commands it seems valid to simply short circuit.

I sense the only other option is to remove the magic auto-detection logic and explicitly register all the commands.